### PR TITLE
Align localizer with planning frame

### PR DESCRIPTION
**Pull request by raybjork**
This PR changes the way MovementSensorLocalizers are used such that they have a local origin from which to calculate distances from.  Additionally it exposes a mechanism with which to calibrate the localizer to east, which is what is necessary for MoveOnGlobe to function as intended.

### CODE REVIEW

File: components/base/kinematicbase/differentialDrive.go [hunk 1]
```diff
@@ -8,22 +8,23 @@ import (
...
 	"go.viam.com/rdk/motionplan"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/services/motion"
 	"go.viam.com/rdk/spatialmath"
+	utils "go.viam.com/utils"
 )
 
 const (
-	distThresholdMM         = 100
+	distThresholdMM         = 1000
```
Comment (biotinker) [inline components/base/kinematicbase/differentialDrive.go:23]: Can we add some comments here discussing what these are used for/what they represent? "distThresholdMM` and `deviationThreshold` look very similar.

File: components/base/kinematicbase/differentialDrive.go [hunk 2]
```diff
@@ -36,44 +37,42 @@ var ErrMovementTimeout = errors.New("movement has timed out")
...
 	}
-
-	var geometry spatialmath.Geometry
 	if len(geometries) > 0 {
-		geometry = geometries[0]
-	}
-	model, err := referenceframe.New2DMobileModelFrame(b.Name().ShortName(), limits, geometry)
-	if err != nil {
-		return nil, err
+		ddk.geometry = geometries[0]
```
Comment (biotinker) [inline components/base/kinematicbase/differentialDrive.go:59]: Can we add on a big ol' WARN log message here if `len(geometries) > 1`?

File: components/base/kinematicbase/differentialDrive.go [hunk 3]
```diff
@@ -36,44 +37,42 @@ var ErrMovementTimeout = errors.New("movement has timed out")
...
+	return ddk, nil
 }
 
 type differentialDriveKinematics struct {
 	base.Base
+	logger                        golog.Logger
 	localizer                     motion.Localizer
-	model                         referenceframe.Model
+	model                         referenceframe.Frame
+	geometry                      spatialmath.Geometry
```
Comment (biotinker) [inline components/base/kinematicbase/differentialDrive.go:75]: Maybe this will become clear as I read the rest of the PR, but doesn't this duplicate the geometry between both this field and the model?

File: components/base/kinematicbase/differentialDrive.go [hunk 4]
```diff
@@ -90,7 +89,8 @@ func (ddk *differentialDriveKinematics) CurrentInputs(ctx context.Context) ([]re
 		return nil, err
 	}
 	pt := pif.Pose().Point()
-	theta := math.Mod(pif.Pose().Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	theta := math.Mod(pif.Pose().Orientation().EulerAngles().Yaw, 2*math.Pi) - math.Pi
```
Comment (biotinker) [inline components/base/kinematicbase/differentialDrive.go:92]: Why this change?

File: services/motion/builtin/builtin.go [hunk 5]
```diff
@@ -217,66 +219,70 @@ func (ms *builtIn) MoveOnGlobe(
...
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(movementSensorName)
+	}
+	origin, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// add an offset between the movement sensor and the base if it is applicable
+	baseOrigin := referenceframe.NewPoseInFrame(componentName.ShortName(), spatialmath.NewZeroPose())
```
Comment (biotinker) [inline services/motion/builtin/builtin.go:268]: Nothing to change for this PR, but I think the string-based (or at least short-name-based) keying for the FS service should be deprecated and replaced with resource names, with a slowly growing urgency.

File: services/motion/builtin/builtin.go [hunk 6]
```diff
@@ -217,66 +219,70 @@ func (ms *builtIn) MoveOnGlobe(
...
+	movementSensorToBase, err := ms.fsService.TransformPose(ctx, baseOrigin, movementSensorName.ShortName(), nil)
+	if err != nil {
+		movementSensorToBase = baseOrigin
+	}
+	ms.logger.Warnf("calibration %v", movementSensorToBase.Pose().Orientation().OrientationVectorDegrees())
+	localizer := motion.NewMovementSensorLocalizer(movementSensor, origin, movementSensorToBase.Pose())
```
Comment (biotinker) [inline services/motion/builtin/builtin.go:274]: IMO this is an awkward place so solve this problem. You're going out of your way to do your own spatial transforms, and it means that the localizer isn't necessarily returning what one might think it's returning.

I would prefer that instead this get added to the framesystem that we construct, and this get set as our goal. Then all the code in `localizer.go` doing `Compose()` calls goes away, and the FS will handle everything.

File: services/motion/localizer.go [hunk 7]
```diff
@@ -2,59 +2,70 @@ package motion
...
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		o = &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: compass}
 	}
-	pose := spatialmath.GeoPointToPose(gp)
-	return *referenceframe.NewPoseInFrame(m.Name().Name, pose), nil
+	pose := spatialmath.NewPose(spatialmath.GeoPointToPose(gp, m.origin).Point(), o)
+	correction := spatialmath.Compose(m.calibration, spatialmath.NewPoseFromOrientation(&spatialmath.OrientationVector{OZ: 1, Theta: -math.Pi / 2}))
```
Comment (biotinker) [inline services/motion/localizer.go:69]: Both this and the Compose below are very unintuitive to follow, and will go away if the calibration is done in the FS as mentioned above.

**Review by biotinker — CHANGES_REQUESTED** [review]
Couple of nits, and one larger architectural question.

File: components/base/kinematicbase/differentialDrive.go (hunk 4 above)
Comment (raybjork) [reply components/base/kinematicbase/differentialDrive.go:107]: Orientation Vectors were exhibiting numerical instability

File: services/motion/builtin/builtin.go (hunk 5 above)
Comment (raybjork) [reply services/motion/builtin/builtin.go:269]: Yeah hopefully we'll have time to do that some time in the near future 😅

File: components/base/kinematicbase/differentialDrive.go (hunk 3 above)
Comment (raybjork) [reply components/base/kinematicbase/differentialDrive.go:90]: no thats a good catch, nice one

File: components/base/kinematicbase/differentialDrive.go (hunk 4 above)
Comment (biotinker) [reply components/base/kinematicbase/differentialDrive.go:107]: Euler angles will have gimbal lock issues that will also show numeric stability; what you want for the Yaw value may actually be in Roll.

The proper way to do this will be to Transform an OV of OY=1 by this orientation, then take the atan2 of the OX and OY values of the resulting OV.

File: components/base/kinematicbase/ptgKinematics.go [hunk 8]
```diff
@@ -111,6 +112,9 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		return errors.New("inputs to ptg kinematic base must be length 3")
 	}
 
+	logger := golog.NewLogger("ddk")
```
Comment (biotinker) [inline components/base/kinematicbase/ptgKinematics.go:115]: I don't think we want to be defining a brand new logger with every GoToInputs call

**Review by biotinker — CHANGES_REQUESTED** [review]
One issue remaining with the orientation of plans, plus some extraneous log messages, otherwise looking good.

File: components/base/kinematicbase/ptgKinematics.go (hunk 8 above)
Comment (raybjork) [reply components/base/kinematicbase/ptgKinematics.go:115]: this was something I did for testing convenience, will fix

File: components/base/kinematicbase/differentialDrive.go (hunk 4 above)
Comment (raybjork) [reply components/base/kinematicbase/differentialDrive.go:107]: They will not have gimbal lock because of the nature of the orientations in question means they will only ever rotate about yaw and therefore are we do not see changes in the domain.  On the contrary Orientation Vectors have a problem where they will dance between the two possible solutions and cause wildly different angles.

Comment (biotinker) [reply components/base/kinematicbase/differentialDrive.go:107]: Not a big fan of euler angles. Agree we won't see gimbal lock during normal operation, but this will break if a vehicle ever points straight up or down. But agree it's better than what currently exists. Can we add a TODO to make this more robust?

**Review by biotinker — APPROVED** [review]
LGTM
